dev_tools/utils/nxdl_utils.py

```diff
@@ -46,7 +46,7 @@ def decode_or_not(elem, encoding: str = "utf-8", decode: bool = True):
         try:
             return elem.decode(encoding)
         except UnicodeDecodeError as e:
-            raise ValueError(f"Error decoding bytes: {e}")
+            raise ValueError(f"Error decoding bytes: {e}") from e
```

Comment:
You don't need to include the exception `e`  in `ValueError`

Reply:
removed